### DISCUSSION TITLE
Database is slowing when pushing new data

### DISCUSSION THREAD

**Discussion by Mitsichury** [appy-one/acebase-server#20] [Q&A] [answered]
Hi everyone, I am interested into this database as it can run as an embedded database.

I was running a benchmark: add 10000 entries in order to know how it reacts with a large amount of data.

I add the data this way:
async addEntry(entry) {
     return await this.db.ref("entry").push(entry);
}

the first entry is added in less than 10ms, after 400 entries the new entry is added in 100ms.

I'm sure that I did not understood how to use it corrrectly, can anyone help me to fix that please ?

Thank you :) !

**Comment by appy-one** [comment, marked as answer]
As a node grows it switches methods to store its child nodes. Up to 100 children are stored linearly, then it switches to an index (binary B+Tree) which is faster to lookup and modify children. Such a tree is optimized for the total number of children currently in it, and allows for up to 10% growth with push id's. Once the tree is full it has to allocate more space to rewrite it with more capacity and another 10% free space. Those tree rewrites are heavy and are probably the reason you see the response time increase at those particular inserts, following inserts should be fast again. As the node grows bigger, it takes longer before such a rewrite has to be executed again (more inserts that fit the 10% free space)

**Reply by Mitsichury** [reply]
Thank you for the explanation, I also added indexes and adding a entry after 10k entries last less than 20ms, it rocks :D !
